lib/jsdom/living/nodes/Element-impl.js

```diff
@@ -430,6 +430,58 @@ class ElementImpl extends NodeImpl {
 
     return attr;
   }
+
+  insertAdjacentHTML(position, text) {
+    position = position.toLowerCase();
+
+    let context;
```

Comment:
So as-is context is unused. I think the spec is not going to be very useful for a guidance on this since it includes "invoking the fragment parsing algorithm with text as markup, and context as the context element" which I don't think we have the ability to do (yet). I made a note of that in https://github.com/tmpvar/jsdom/pull/1316#issuecomment-172030345

Instead I think you should remove context, and just check this.parentNode in line 442. Lines 542 and 460-462 can go away.